tests/data_sources/test_rasterdatasource.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+
+import pytest
+import xarray as xr
+from pydantic import ValidationError
+
+from hydromt.data_sources import RasterDataSource
+from hydromt.drivers.rasterdataset_driver import RasterDatasetDriver
+from hydromt.metadata_resolvers import MetaDataResolver
+
+
+class TestRasterDataSource:
+    @pytest.fixture()
+    def example_source(
+        self,
+        mock_rasterds_driver: RasterDatasetDriver,
+        mock_resolver: MetaDataResolver,
+        tmp_dir: Path,
+    ):
+        return RasterDataSource(
+            root=".",
+            name="example_rasterds",
+            driver=mock_rasterds_driver,
+            metadata_resolver=mock_resolver,
+            uri=str(tmp_dir / "rasterds.zarr"),
```

Comment:
not a fan of this syntax. I couldn't find when it was added, so if you want to keep it please make ure it works from 3.9 onwards.

Reply:
This pathlib, so a python `Path` works with `Path("/path/to") / file`

Reply:
Then it's okay!